tachyon/zk/base/entities/prover.h

```diff
@@ -41,21 +41,22 @@ class Prover : public Entity<PCSTy> {
     return this->domain_->size() - (blinder_.blinding_factors() + 1);
   }
 
-  bool CommitEvals(const Evals& evals) {
+  [[nodiscard]] bool CommitEvals(const Evals& evals) {
     if (evals.NumElements() != this->domain_->size()) return false;
 
     Commitment commitment;
     if (!this->pcs_.CommitLagrange(evals, &commitment)) return false;
     return GetWriter()->WriteToProof(commitment);
   }
 
-  bool CommitEvalsWithBlind(const Evals& evals, BlindedPolynomial<Poly>* out) {
+  [[nodiscard]] bool CommitEvalsWithBlind(const Evals& evals,
+                                          BlindedPolynomial<Poly>* out) {
     if (!CommitEvals(evals)) return false;
     *out = {this->domain_->IFFT(evals), blinder_.Generate()};
     return true;
   }
 
-  bool Evaluate(const Poly& poly, const F& x) {
+  [[nodiscard]] bool Evaluate(const Poly& poly, const F& x) {
```

Comment:
It would be better to include the change to this line in the previous commit.

Reply:
I'll just squash the commits!